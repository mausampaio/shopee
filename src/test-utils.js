import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";

export const mockProduct = {
  id: 1,
  name: "Produto de teste",
  price: 1.0,
  description: "Produto gerado para testes",
  url: "produto-de-teste",
  image: "./produto-teste.jpg",
};

export const mockCart = {
  id: mockProduct.id,
  name: mockProduct.name,
  price: mockProduct.price,
  quantity: 10,
};

export const mockContext = {
  cart: [],
  products: [],
  product: {},
  loading: false,
  getCartTotal: () => 0,
};

const RouterProvider = ({ children }) => {
  return <BrowserRouter>{children}</BrowserRouter>;
};

const customRender = (ui, options) => {
  return render(<RouterProvider>{ui}</RouterProvider>, { ...options });
};

export * from "@testing-library/react";

export { customRender as render };
