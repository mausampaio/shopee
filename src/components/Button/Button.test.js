import { render, fireEvent } from "../../test-utils";

import { Button } from "./index";

test("should match snapshot", () => {
  const { asFragment } = render(<Button>Button</Button>);
  expect(asFragment()).toMatchSnapshot();
});

test("should call an onClick event when pressed", () => {
  const handleClick = jest.fn();
  const { getByRole } = render(<Button onClick={handleClick} />);
  fireEvent.click(getByRole("button"));
  expect(handleClick).toBeCalledTimes(1);
});
