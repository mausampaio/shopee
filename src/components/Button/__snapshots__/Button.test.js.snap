// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  .c0 {
  width: 100%;
  min-width: 44px;
  height: 44px;
  background-color: var(--secondary);
  border-radius: 4px;
  color: var(--primary);
  font-size: 18px;
  font-weight: 500;
  cursor: pointer;
}

<button
    class="c0"
  >
    Button
  </button>
</DocumentFragment>
`;
